src/components/atoms/ButtonNG/ButtonNG.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { CSSProperties } from "react";
+import { Link } from "react-router-dom";
+import classNames from "classnames";
+
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+import { faCircleNotch } from "@fortawesome/free-solid-svg-icons/faCircleNotch";
+
+import "./ButtonNG.scss";
+
+export type ButtonGradientType =
+  | "gradient"
+  | "admin-gradient"
+  | "danger-gradient";
+
+export type ButtonType = "button" | "reset" | "submit";
+export type ButtonVariant = "primary" | "secondary" | "white" | "dark";
+export type ButtonIconSize = "1x" | "2x" | "3x";
+
+export interface ButtonProps {
+  className?: string;
+  style?: CSSProperties;
+  disabled?: boolean;
+  onClick?: () => void;
+  isLink?: boolean;
+  linkTo?: string;
+  newTab?: boolean;
+  loading?: boolean;
+  gradient?: ButtonGradientType;
+  type?: ButtonType;
+  variant?: ButtonVariant;
+  iconOnly?: boolean;
+  iconName?: IconProp;
+  iconSize?: ButtonIconSize;
+}
+
+export const ButtonNG: React.FC<ButtonProps> = ({
+  children,
+  className = "",
+  style,
+  loading,
+  onClick,
+  type = "button",
+  gradient = "",
+  isLink = false,
+  linkTo = "",
+  disabled = false,
+  newTab = false,
+  variant = "",
+  iconOnly = false,
+  iconName,
+  iconSize = "1x",
+}) => {
+  const parentClasses = classNames({
+    "ButtonNG ButtonNG__link": isLink,
+    "ButtonNG ButtonNG__button": !isLink,
+    "ButtonNG--disabled": disabled,
+    "ButtonNG--enabled": !disabled,
+    "ButtonNG--loading": loading,
+    [`ButtonNG--icon-only ButtonNG--${iconSize}`]: iconOnly,
+    [`ButtonNG--icon-text`]: !iconOnly,
+    [`ButtonNG--${gradient}`]: gradient && !disabled,
+    [`ButtonNG--${variant}`]: variant && !disabled,
+    [className]: className,
+  });
```

Comment:
```suggestion
  const parentClasses = classNames(className, {
    "ButtonNG ButtonNG__link": isLink,
    "ButtonNG ButtonNG__button": !isLink,
    "ButtonNG--disabled": disabled,
    "ButtonNG--enabled": !disabled,
    "ButtonNG--loading": loading,
    [`ButtonNG--icon-only ButtonNG--${iconSize}`]: iconOnly,
    [`ButtonNG--icon-text`]: !iconOnly,
    [`ButtonNG--${gradient}`]: gradient && !disabled,
    [`ButtonNG--${variant}`]: variant && !disabled,
  });
```

Reply:
No explanation provided why `[className]: className,` should be removed.

Reply:
`classNames` function is already accept `className` as the first argument. It's tastier to add `classNames(className, {...}` rather then `[className]: className`

WDYT?

Reply:
The issue is this: If you didn't point it out to me, I wouldn't have seen it.

That's the reason why I usually just don't mix string + object properties but either go with one or the other.

Reply:
If no objection, I will leave it as is.